src/js/components/AppListComponent.jsx

```diff
@@ -70,13 +70,19 @@ var AppListComponent = React.createClass({
     var sortKey = state.sortKey;
     var props = this.props;
 
-    return lazy(state.apps)
+    var appsSequence = lazy(state.apps);
+
+    if (props.filterText != null && props.filterText != "") {
+      appsSequence = appsSequence
+        .filter(function (app) {
+          return app.id.indexOf(props.filterText) !== -1;
```

Comment:
We could even go as far using `lazy().indexOf` imho. http://danieltao.com/lazy.js/docs/#Sequence-indexOf